src/hardware/pcspeaker.cpp

```diff
@@ -493,7 +493,11 @@ static inline void add_impulse(const float index, const float amplitude)
 
 static void PCSPEAKER_CallBack(uint16_t len)
 {
-	int16_t *stream = (int16_t *)MixTemp;
+	static std::vector<int16_t> mix_buffer;
+	if (mix_buffer.size() < len)
+		mix_buffer.resize(len);
```

Comment:
Minor: can't you just simply always call `resize` and the container will figure out the rest? (it will be a no-op if the size is unchanged anyway)

Reply:
Yup!
(in a later commit, I drop the vector which required shuffling played samples forward in favor of the deque, which is more like a circular buffer and doesn't require the shuffling.